src/components/transport_manager/src/transport_manager_impl.cc

```diff
@@ -540,14 +540,19 @@ void TransportManagerImpl::AddConnection(const ConnectionInternal& c) {
   connections_.push_back(c);
 }
 
-void TransportManagerImpl::RemoveConnection(uint32_t id) {
+void TransportManagerImpl::RemoveConnection(
+    uint32_t id, transport_adapter::TransportAdapter* transport_adapter) {
```

Comment:
In the message queue, where the method is called, the TransportAdapter is a pointer, so I preferred to keep this form. There is no restriction that it should be not NULL so I added a check in the method. Also according to the  [(Google C++ style)](https://google.github.io/styleguide/cppguide.html#Reference_Arguments) it is an output parameter so should be a pointer. Please correct me if I am wrong.